### DISCUSSION TITLE
How to continue same conversation

### DISCUSSION THREAD

**Discussion by omkar-tenkale** [chatwoot/chatwoot#3250] [Q&A]
I'm using chatwoot in my android app

As there is no official SDK yet, I've created a help.html static file and loading the url in webview
file:///android_asset/onlinehelp.html

i'm also setting userId and email so all conversations will be tied to a user

Problem is,
When user navigaets to help screen and url is loaded and he enters a message, if support is not online he'll navigate back

Next time he comes back to check for any replies, a new conversation is started and he cant see replies to the old conversation

My question is how to configure sdk in javascript to continue the old conversation?

**Comment by omkar-tenkale** [comment]
if i host it on a domain, it works as expected
It seems an issue with file:/// uri ?

**Comment by sojan-official** [comment]
@omkar-tenkale  have you taken a look at 
https://github.com/chatwoot/chatwoot-react-native-widget
https://github.com/chatwoot/chatwoot-flutter-sdk

**Reply by omkar-tenkale** [reply]
@sojan-official  I cant use that as i'm building my app natively with java and kotlin

**Reply by pranavrajs** [reply]
@omkar-tenkale Have you used Identity validation? This would keep the previous conversation thread irrespective of the session.  https://www.chatwoot.com/docs/product/channels/live-chat/sdk/setup#identity-validation-using-hmac

**Reply by omkar-tenkale** [reply]
@pranavrajs  Yes, I'm doing it but i guess the problem is chatwoot saves session in cookies and browsers dont store cookies for file:///  url

**Reply by hemantgoyal** [reply]
Hi. We are facing the same issue. Is identity validation using hmac enough? 
We are using a self hosted Chatwoot. We are facing a chat persistance issue, we are creating a contact using the API, and then a conversation for the contact using the API as well. An identifier is passed while creating the contact, and an hash based on the USER Identity Validation is also created to persist with the conversation. Now when we try to add the Hash to the widget, it still creates a new conversation over continuing with the existing one? Is there anyway the widget can be pointed to the existing conversation? I can only see setting up the user in the widget and the identity hash.
